### ISSUE TITLE
ENH: support GeoSeries.z alongside x and y

### ISSUE THREAD

**Issue by AlexanderJuestel** [geopandas/geopandas#1765]
- [ ] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

#### Code Sample, a copy-pastable example

```ruby

from shapely.geometry import Point

point = Point(1,2,3)

import geopandas as gpd

gdf = gpd.GeoDataFrame(geometry=[point, point])

gdf.geometry.z

~\Anaconda3\envs\test_gempy\lib\site-packages\pandas\core\generic.py in __getattr__(self, name)
   5137             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5138                 return self[name]
-> 5139             return object.__getattribute__(self, name)
   5140 
   5141     def __setattr__(self, name: str, value) -> None:

AttributeError: 'GeoSeries' object has no attribute 'z'

```

#### Problem description

I created a GeoDataFrame consisting of Shapely Points that also have a Z component. The X and Y coordinates can easily be obtained using `gdf.geometry.x` or `gdf.geometry.y` but `gdf.geometry.z` returns the Attribute error above.

The z coordinates can be accessed though using `gdf.geometry.loc[0].z` but to be consistent and to avoid loops `gdf.geometry.z` should also work. 

#### Expected Output

`gdf.geometry.z`
0   3.00
1   3.00
dtype: float64

Thanks for your responses, maybe there is also just another simple way that I just have not found yet

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.8.5 (default, Sep  3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)]
executable : C:\Users\...\Anaconda3\envs\test_gempy\python.exe
machine    : Windows-10-10.0.19041-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.1.2
GDAL data dir: None
PROJ       : 7.0.0
PROJ data dir: C:\Users\...\Anaconda3\envs\...\Library\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.8.1
pandas     : 1.1.3
fiona      : 1.8.17
numpy      : 1.19.2
shapely    : 1.7.1
rtree      : 0.9.4
pyproj     : 2.6.1.post1
matplotlib : 3.3.2
mapclassify: None
geopy      : 2.0.0
psycopg2   : None
geoalchemy2: None
pyarrow    : None

</details>

**Comment by martinfleis** [comment]
Hi, thanks for the report.

This is not necessarily a bug, it is just not implemented. However, it should be simple to do. We just need to copy the implementation of `x` and `y` and delegate `z` property in a similar manner. Which means a new property like this one

https://github.com/geopandas/geopandas/blob/36521787c3e6484c00c2ffb8cdde60798d6d1616/geopandas/geoseries.py#L238

and implementation on the array level here 
https://github.com/geopandas/geopandas/blob/36521787c3e6484c00c2ffb8cdde60798d6d1616/geopandas/array.py#L703-L709
and here
https://github.com/geopandas/geopandas/blob/36521787c3e6484c00c2ffb8cdde60798d6d1616/geopandas/_vectorized.py#L857-L861

Feel free to give it a try! (or anyone else)

**Comment by AlexanderJuestel** [comment]
Hey,

I'll give it a try :)

Cheers Alex

**Comment by vangorade** [comment]
Hi, If you don't mind can i give it a try @AlexanderJuestel  it would be my great first contribution to this project :)

**Comment by AlexanderJuestel** [comment]
Sure, go ahead @vangorade. I am busy with work these days :)

**Comment by vangorade** [comment]
sure thanks @AlexanderJuestel !

**Comment by vangorade** [comment]
take

**Comment by raybellwaves** [comment]
Can this be closed?

**Comment by martinfleis** [comment]
Closed by #1773 (thanks @raybellwaves for a notice)
